src/jpl/dijit/GranulesController.js

```diff
@@ -620,17 +621,132 @@ define([
+                url += "&temporal[]=" + DOMUtil.prototype.dateFormatISOBeginningOfDay(this.startDateWidget.get("value")) + "," + DOMUtil.prototype.dateFormatISOEndOfDay(this.endDateWidget.get("value"));
+                url += "&sort_key[]=" + (this.granuleGrid._sort[0].descending ? "-" : "%2B") + sort
+                
+                if(this.nameFilterBox.get("value")){
+                    url += "&native_id[]=" + this.nameFilterBox.get("value") + "&options[native_id][pattern]=true";
+                }
+
+                withCredentials = this.config.hitide.externalConfigurables.crossOriginCmrCookies;
+            }
+            else{
+                url = this.config.hitide.externalConfigurables.granuleSearchService + "?";
+                url += "datasetId=" + this.datasetId;
+                url += "&startTime=" + DOMUtil.prototype.dateFormatISOBeginningOfDay(this.startDateWidget.get("value"));
+                url += "&endTime=" + DOMUtil.prototype.dateFormatISOEndOfDay(this.endDateWidget.get("value"));
+                url += "&name=" + this.nameFilterBox.get("value");
+                url += "&bbox=" + (this.bbox || "");
+                url += "&itemsPerPage=" + this.availableGranules;
+                url += "&startIndex=" + this.currentSolrIdx;
+                url += "&sort=" + this.granuleGrid._sort[0].attribute + (this.granuleGrid._sort[0].descending ? " desc" : " asc");
+            }
+
+            var _context = this;
+            var r;
+            var topicHandler = topic.subscribe(SearchEvent.prototype.CANCEL_REQUESTS, function(message) {
+                if (message.target === "*" || message.target === this.datasetId) {
+                    r.cancel();
+                }
+            });
+            r = xhr.get(url, {
+                headers: {
+                    "X-Requested-With": null
+                },
+                handleAs: "json",
+                method: "get",
+                withCredentials: withCredentials
+            }).then(function(response) {
+                // Unsubscribe from cancel requests
+                topicHandler.remove();
+                // Update accordingly
+                if (_context.domNode) {
+                    var granuleNamesToReturn
+                    if (_context.source === 'cmr') {
+                        granuleNamesToReturn = response.items.map(function(granuleObj) {
+                            return granuleObj["meta"]["native-id"];
+                        });
+                    } else {
+                        granuleNamesToReturn = response.response.docs.map(function(granuleObj) {
+                            return granuleObj['Granule-Name']
+                        })
+                    }
+                    var startDate = moment.utc(_context.startDateWidget.getValue());
+                    var endDate = moment.utc(_context.endDateWidget.getValue());
+                    var queryId = Math.floor(Math.random() * (9999999999 - 0)) + 0;
```

Comment:
What's going on here? Is this like a javascript thing to get an integer from math.random?

Reply:
I am not sure what this is doing exactly. I just moved this code to a different part of the file.